tokio/src/net/tcp/listener.rs

```diff
@@ -94,6 +96,7 @@ impl TcpListener {
     ///     Ok(())
     /// }
     /// ```
+    #[cfg(not(target_os = "wasi"))]
     pub async fn bind<A: ToSocketAddrs>(addr: A) -> io::Result<TcpListener> {
```

Comment:
All of the constructors in this file appear to have been disabled. How do we create a TcpListener?

Reply:
These have to be created on behalf of the WebAssembly application by the WebAssembly runtime, which then provides the socket file descriptor to the Wasm application.

Reply:
@Darksonn To answer that question more directly: `FromRawFd`. The WebAssembly runtime pre-creates the initial file descriptors and passes them to the module. There is not currently any way to dynamically bind.

Reply:
Same here, I would consider making a `cfg_not_wasi` { ... }` macro